[CI] SLMSnapshotBlockingIntegTests testRetentionWithMultipleRepositories failing

This is a newly added test that has already failed a number of times in CI.

**Build scan**:
https://gradle-enterprise.elastic.co/s/ffmncmzt23eay/tests/:x-pack:plugin:ilm:internalClusterTest/org.elasticsearch.xpack.slm.SLMSnapshotBlockingIntegTests/testRetentionWithMultipleRepositories#1

**Repro line**:
`./gradlew ':x-pack:plugin:ilm:internalClusterTest' --tests "org.elasticsearch.xpack.slm.SLMSnapshotBlockingIntegTests.testRetentionWithMultipleRepositories" -Dtests.seed=9BFBC3AF7F9C991C -Dtests.security.manager=true -Dtests.locale=es-DO -Dtests.timezone=SST -Druntime.java=11`

**Reproduces locally?**:
Nope.

**Applicable branches**:
7.11, 7.12, 7.x and master

**Failure history**:
https://gradle-enterprise.elastic.co/scans/tests?search.relativeStartTime=P7D&search.timeZoneId=America/Los_Angeles&tests.container=org.elasticsearch.xpack.slm.SLMSnapshotBlockingIntegTests&tests.disabledDistributions=WyJvdXRjb21lOnBhc3NlZCIsIm91dGNvbWU6c2tpcHBlZCJd&tests.sortField=FAILED&tests.test=testRetentionWithMultipleRepositories&tests.unstableOnly=true

**Failure excerpt**:
```
org.elasticsearch.xpack.slm.SLMSnapshotBlockingIntegTests > testRetentionWithMultipleRepositories FAILED
    java.lang.AssertionError: timed out waiting for green state
        at __randomizedtesting.SeedInfo.seed([9BFBC3AF7F9C991C:35EAB13BB52B8935]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.elasticsearch.test.ESIntegTestCase.ensureColor(ESIntegTestCase.java:903)
        at org.elasticsearch.test.ESIntegTestCase.ensureGreen(ESIntegTestCase.java:856)
        at org.elasticsearch.test.ESIntegTestCase.ensureGreen(ESIntegTestCase.java:845)
        at org.elasticsearch.xpack.slm.SLMSnapshotBlockingIntegTests.createAndPopulateIndex(SLMSnapshotBlockingIntegTests.java:465)
        at org.elasticsearch.xpack.slm.SLMSnapshotBlockingIntegTests.testUnsuccessfulSnapshotRetention(SLMSnapshotBlockingIntegTests.java:349)
        at org.elasticsearch.xpack.slm.SLMSnapshotBlockingIntegTests.testRetentionWithMultipleRepositories(SLMSnapshotBlockingIntegTests.java:280)
```

--- a/x-pack/plugin/ilm/src/internalClusterTest/java/org/elasticsearch/xpack/slm/SLMSnapshotBlockingIntegTests.java
+++ b/x-pack/plugin/ilm/src/internalClusterTest/java/org/elasticsearch/xpack/slm/SLMSnapshotBlockingIntegTests.java
@@ -17,6 +17,7 @@ import org.elasticsearch.action.admin.cluster.snapshots.status.SnapshotsStatusRe
 import org.elasticsearch.action.search.SearchResponse;
 import org.elasticsearch.cluster.SnapshotsInProgress;
 import org.elasticsearch.cluster.health.ClusterHealthStatus;
+import org.elasticsearch.cluster.routing.UnassignedInfo;
 import org.elasticsearch.common.Strings;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.common.unit.TimeValue;
@@ -267,7 +268,6 @@ public class SLMSnapshotBlockingIntegTests extends AbstractSnapshotIntegTestCase
         testUnsuccessfulSnapshotRetention(true);
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/70185")
     public void testRetentionWithMultipleRepositories() throws Exception {
         disableRepoConsistencyCheck("test leaves behind an empty repository");
         final String secondRepo = "other-repo";
@@ -278,6 +278,11 @@ public class SLMSnapshotBlockingIntegTests extends AbstractSnapshotIntegTestCase
                 true, new SnapshotRetentionConfiguration(null, 1, 2));
         logger.info("-->  start snapshot");
         client().execute(ExecuteSnapshotLifecycleAction.INSTANCE, new ExecuteSnapshotLifecycleAction.Request(policyId)).get();
+        // make sure the SLM history data stream is green and won't not be green for long because of delayed allocation when data nodes
+        // are stopped
+        ensureGreen(SLM_HISTORY_DATA_STREAM);
+        client().admin().indices().prepareUpdateSettings(SLM_HISTORY_DATA_STREAM).setSettings(
+                Settings.builder().put(UnassignedInfo.INDEX_DELAYED_NODE_LEFT_TIMEOUT_SETTING.getKey(), 0)).get();
         testUnsuccessfulSnapshotRetention(randomBoolean());
     }
 

